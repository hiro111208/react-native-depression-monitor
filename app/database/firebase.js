//import firebase from "firebase/app";
import "firebase/firestore";
import firebase from 'firebase'
require('firebase/auth')

const firebaseConfig = {
  apiKey: "AIzaSyDwOUkdxC7kNNILoFTSx05PTUxRci_kcaw",
  authDomain: "fireship-tutorial-646fc.firebaseapp.com",
  projectId: "fireship-tutorial-646fc",
  storageBucket: "fireship-tutorial-646fc.appspot.com",
  messagingSenderId: "289064122528",
  appId: "1:289064122528:web:ee39abfa2183c8499071c5",
  measurementId: "G-D35RDH9V0E",
};

firebase.initializeApp(firebaseConfig);

export default firebase;
