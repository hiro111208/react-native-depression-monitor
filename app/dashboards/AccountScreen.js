import React, { useState } from "react";
import { View, Text, StyleSheet, TouchableOpacity, Image } from "react-native";
import colors from "../config/colors";
import firebase from "../database/firebase";
import * as indexStyles from "../config/indexStyles";

/**
 * Screen that renders account options such as signing out,
 * replaying demo or navigating to support resources
 */
export default function AccountScreen({ props, navigation }) {
    const [displayName, setDisplayName] = useState(
        firebase.auth().currentUser !== null
            ? firebase.auth().currentUser.displayName
            : ""
    );

    // Sign out from the application
    const signOut = () => {
        firebase
            .auth()
            .signOut()
            .then(() => {
                console.log("Logout successful");
                navigation.popToTop();
            })
            .catch((error) => setErrorMessage(error.message));
    };

    return (
        <View
            style={[styles.container, indexStyles.centering, indexStyles.cover]}
        >
            <View
                style={[
                    styles.center,
                    indexStyles.shadowEffect,
                    indexStyles.cover,
                ]}
            >
                <View style={{ height: "5%" }}></View>

                {/* Render logo of the app */}
                <View style={[{ height: "40%" }, indexStyles.centering]}>
                    <Image
                        style={{ width: 200, height: 200 }}
                        resizeMode="contain"
                        source={require("../assets/hand-logo.png")}
                        testID={"LOGO"}
                    />
                </View>

                {/* Support resources button */}
                <View style={[{ height: "10%" }]}>
                    <TouchableOpacity
                        onPress={() => navigation.navigate("SupportResources")}
                        style={[
                            styles.optButton,
                            indexStyles.cover,
                            indexStyles.centering,
                        ]}
                        testID={"SUPPORT_BUTTON"}
                    >
                        <Text style={[styles.fontStyle, { fontSize: 17 }]}>
                            Support Resources
                        </Text>
                    </TouchableOpacity>
                </View>

                <View style={{ height: "2%" }}></View>

                {/* Button to demo screen */}
                <View style={[{ height: "10%" }]}>
                    <TouchableOpacity
                        onPress={() => navigation.navigate("DemoScreen")}
                        style={[
                            styles.optButton,
                            indexStyles.cover,
                            indexStyles.centering,
                        ]}
                        testID={"DEMO_BUTTON"}
                    >
                        <Text style={[styles.fontStyle, { fontSize: 17 }]}>
                            Replay Demo
                        </Text>
                    </TouchableOpacity>
                </View>

                <View style={{ height: "23%" }}></View>

                {/* Sign out button that goes back to login screen */}
                <View style={[{ height: "10%" }, indexStyles.centering]}>
                    <TouchableOpacity
                        onPress={() => signOut()}
                        style={[
                            styles.optButton,
                            indexStyles.cover,
                            indexStyles.centering,
                            {
                                borderBottomLeftRadius: 40,
                                borderBottomRightRadius: 40,
                            },
                        ]}
                        testID={"LOGOUT_BUTTON"}
                    >
                        <Text
                            style={[
                                styles.fontStyle,
                                { fontSize: 17, color: "black" },
                            ]}
                        >
                            LOG OUT
                        </Text>
                    </TouchableOpacity>
                </View>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    center: {
        backgroundColor: colors.mainPanel,
        borderRadius: 50,
        borderWidth: 5,
        borderColor: colors.lightOutline,
    },
    container: {
        flex: 1,
        display: "flex",
        padding: 25,
        backgroundColor: "#fff",
    },
    fontStyle: {
        fontWeight: "bold",
        color: "dimgray",
    },
    optButton: {
        backgroundColor: colors.lightOutline,
    },
});
