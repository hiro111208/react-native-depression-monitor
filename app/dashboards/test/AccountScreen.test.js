import React from "react";
import { shallow } from "enzyme";
import { render, fireEvent, cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import AccountScreen from "../AccountScreen";
import firebase from "../../database/firebase";

describe("Testing AccountScreen.js rendering", () => {
  
  it("AccountScreen renders correctly", () => {
    const wrapper = shallow(<AccountScreen />);
    expect(wrapper.length).toBe(1);
  });
    
  it("AccountScreen snapshot", () => {
    const tree = renderer.create(<AccountScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
    
  it("Renders logo", () => {
    const wrapper = render(<AccountScreen />);
    expect(wrapper.queryByTestId("LOGO")).toBeTruthy();
  });
    
  it("Renders support resources button", () => {
    const wrapper = render(<AccountScreen />);
    expect(wrapper.queryByTestId("SUPPORT_BUTTON")).toBeTruthy();
  });
    
  it("Renders replay demo button", () => {
    const wrapper = render(<AccountScreen />);
    expect(wrapper.queryByTestId("DEMO_BUTTON")).toBeTruthy();
  });

  it("Renders logout button", () => {
    const wrapper = render(<AccountScreen />);
    expect(wrapper.queryByTestId("LOGOUT_BUTTON")).toBeTruthy();
  });

});

describe("Testing AccountScreen.js navigation", () => {
  
  const navigation = { navigate: jest.fn() };

  it("Navigate to SupportResources", () => {
    const wrapper = render(<AccountScreen navigation={ navigation } />);
    const button = wrapper.getByTestId("SUPPORT_BUTTON");
    expect(navigation.navigate).not.toHaveBeenCalledWith("SupportResources");
    fireEvent(button, "press");
    expect(navigation.navigate).toHaveBeenCalledWith("SupportResources");
  });

  it("Navigate to demo screen", () => {
    const wrapper = render(<AccountScreen navigation={ navigation } />);
    const button = wrapper.getByTestId("DEMO_BUTTON");
    expect(navigation.navigate).not.toHaveBeenCalledWith("DemoScreen");
    fireEvent(button, "press");
    expect(navigation.navigate).toHaveBeenCalledWith("DemoScreen");
  });

});

describe("Test logging a user out", () => {
  const signOut = jest.fn(() => Promise.resolve())

  jest.spyOn(firebase, "auth").mockImplementation(() => {
    return {
      currentUser: {
        displayName: "testUser",
      },
      signOut,
    }
  })

  it("Log out of account", () => {
    const wrapper = render(<AccountScreen />);
    const button = wrapper.getByTestId("LOGOUT_BUTTON");
    
    fireEvent(button, "press");
    expect(signOut).toHaveBeenCalled();
  });

});

afterEach(cleanup);