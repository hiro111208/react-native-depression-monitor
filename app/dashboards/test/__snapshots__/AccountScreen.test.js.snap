// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AccountScreen.js rendering AccountScreen snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "display": "flex",
        "flex": 1,
        "padding": 25,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "height": "100%",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fed8b1",
          "borderColor": "#ffeed2",
          "borderRadius": 50,
          "borderWidth": 5,
        },
        Object {
          "elevation": 5,
          "marginVertical": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "height": "5%",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "height": "40%",
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../app/assets/hand-logo.png",
          }
        }
        style={
          Object {
            "height": 200,
            "width": 200,
          }
        }
        testID="LOGO"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": "10%",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffeed2",
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="SUPPORT_BUTTON"
      >
        <Text
          style={
            Array [
              Object {
                "color": "dimgray",
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 17,
              },
            ]
          }
        >
          Support Resources
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": "2%",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "height": "10%",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffeed2",
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="DEMO_BUTTON"
      >
        <Text
          style={
            Array [
              Object {
                "color": "dimgray",
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 17,
              },
            ]
          }
        >
          Replay Demo
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": "23%",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "height": "10%",
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffeed2",
            "borderBottomLeftRadius": 40,
            "borderBottomRightRadius": 40,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="LOGOUT_BUTTON"
      >
        <Text
          style={
            Array [
              Object {
                "color": "dimgray",
                "fontWeight": "bold",
              },
              Object {
                "color": "black",
                "fontSize": 17,
              },
            ]
          }
        >
          LOG OUT
        </Text>
      </View>
    </View>
  </View>
</View>
`;
